tftb/processing/base.py

```diff
@@ -209,15 +212,17 @@ def plot(self, ax=None, kind='cmap', show=True, default_annotation=True,
                     ax.set_zlabel("Amplitude")
             elif kind == "wireframe":
                 from mpl_toolkits.mplot3d import Axes3D  # NOQA
-                ax = fig.add_subplot(projection="3d")
+                if ax is None:
+                    ax = fig.add_subplot(projection="3d")
                 x = np.arange(self.signal.shape[0])
                 y = np.linspace(0, 0.5, self.signal.shape[0])
                 X, Y = np.meshgrid(x, y)
                 ax.plot_wireframe(X, Y, np.abs(self.tfr), cmap=plt.cm.jet,
                                   rstride=3, cstride=3)
             else:
-                t, f = np.meshgrid(self.ts, np.linspace(0, 0.5, self.tfr.shape[0]))
-                ax = fig.add_subplot(111)
+                t, f = np.meshgrid(self.timestamps, np.linspace(0, 0.5, self.tfr.shape[0]))
+                if ax is None:
+                    ax = fig.add_subplot(111)
```

Comment:
Thanks, the plotting code did indeed need these changes. Whatever else happens to this PR, this part is definitely needed :)